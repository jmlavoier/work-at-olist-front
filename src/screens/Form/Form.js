import React from 'react';
import Box from 'components/Box';

import Button from 'components/Button';
import InputText from 'components/InputText';
import PasswordStrength from 'components/PasswordStrength';
import styles from './Form.sass';

const App = () => (
  <div>
    <Box>
      <div className={styles.subtitle}>
        <h2>Crie sua conta</h2>
      </div>
      <InputText label="Nome completo" onChange={() => {}} value="" />
      <InputText label="E-mail" onChange={() => {}} value="" />
      <PasswordStrength onChange={() => {}} value="" />
      <InputText type="password" label="Confirme sua senha" onChange={() => {}} value="" />
      <div className={styles.footer}>
        <Button onClick={() => {}} >Criar conta</Button>
      </div>
    </Box>
  </div>
);

export default App;
