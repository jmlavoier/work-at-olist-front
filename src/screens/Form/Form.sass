@import 'styles/variables'

.subtitle
  text-align: center

  h2
    margin: 20px 0

.footer
  margin-top: 20px