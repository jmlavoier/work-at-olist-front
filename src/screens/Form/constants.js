export const INITIAL_FIELD = {
  value: '',
  isValid: false,
};

export default INITIAL_FIELD;
