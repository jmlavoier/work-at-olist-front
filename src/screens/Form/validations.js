const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

export const defaultValidation = (value, prevIsValid) => prevIsValid;
export const nameValidation = value => value.length > 0;
export const emailValidation = value => emailRegex.test(value);
export const confirmapassValidation = (passwordValue, value) => passwordValue === value;
