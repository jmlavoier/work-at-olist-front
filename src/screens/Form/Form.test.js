import React from 'react';
import { shallow } from 'enzyme';

import Form from './Form';

describe('<Form />', () => {
  it('Should component match snapshot', () => {
    const wrapper = shallow(<Form />);

    expect(wrapper).toMatchSnapshot();
  });
});
