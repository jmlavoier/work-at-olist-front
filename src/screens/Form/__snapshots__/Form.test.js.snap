// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Should component match snapshot 1`] = `
<div>
  <Box>
    <div
      className="subtitle"
    >
      <h2>
        Crie sua conta
      </h2>
    </div>
    <InputText
      isValid={true}
      label="Nome completo"
      onChange={[Function]}
      value=""
    />
    <InputText
      isValid={true}
      label="E-mail"
      onChange={[Function]}
      value=""
    />
    <PasswordStrength
      onChange={[Function]}
      value=""
    />
    <InputText
      isValid={true}
      label="Confirme sua senha"
      onChange={[Function]}
      type="password"
      value=""
    />
    <div
      className="footer"
    >
      <Button
        isLoading={false}
        onClick={[Function]}
      >
        Criar conta
      </Button>
    </div>
  </Box>
</div>
`;
